commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeDefinition.java

```diff
@@ -82,6 +90,14 @@ public boolean isGlobal() {
       return global;
    }
 
+   public boolean isDeprecated() {
+      return (deprecatedMajor < Integer.MAX_VALUE);
```

Comment:
:thinking:

Reply:
By default, attributes are non-deprecated and have `deprecatedMajor = Integer.MAX_VALUE`.